server/odc-service/src/main/java/com/oceanbase/odc/metadb/regulation/ruleset/MetadataEntity.java

```diff
@@ -99,20 +99,24 @@ public boolean equals(Object obj) {
             return false;
         }
         MetadataEntity that = (MetadataEntity) obj;
-        return Objects.equals(this.getId(), that.getId()) && Objects.equals(this.getBuiltIn(), that.getBuiltIn())
```

Comment:
why don't we use `@EqualAndHashCode`, this class looks like a normal pojo class

Reply:
`MetadataEntity` contains a list of `PropertyMetadataEntity`, and what we need here is that if two lists of  `PropertyMetadataEntity` are the same list regardless of the order of the elements, then the two `MetadataEntity` entities are equal. I use CollectionUtils.isEqualCollection in the `equals` method to judge if these two `PropertyMetadataEntity` lists are equal. I doubt if `@EqualAndHashCode` could handle this scene.

Reply:
can we use `Set< PropertyMetadataEntity>` to regardless the sequence of the `PropertyMetadataEntity`?

Reply:
Yes, using `Set` instead of `List` is a better choice here, but we need to change the `PropertyMetadataEntity`'s `defaultValues` and `candidates` from a list to set as well, and this modification may cause some other changes to the original code, so I prefer not to make this modification here.